internal/database/db.go

```diff
@@ -137,20 +137,30 @@ func (dbs *SQLdb) checkAndReconnectIfNeeded() {
 
 // GetHeader retrieves the file header
 func (dbs *SQLdb) GetHeader(fileID int) ([]byte, error) {
+	r, err := dbs.getHeader(fileID)
+	count := 0
+
+	for err != nil && count < dbRetryTimes {
+		r, err = dbs.getHeader(fileID)
+		count++
+	}
```

Comment:
I would have done it like this to get rid of the first DB call.

```suggestion
	r, err, count := nil, nil, 0

	for ( r == nil || err != nil) && count < dbRetryTimes {
		r, err = dbs.getHeader(fileID)
		count++
	}
```